app/src/main/java/com/concordia/mcga/activities/SplashActivity.java

```diff
@@ -0,0 +1,37 @@
+package com.concordia.mcga.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+
+public class SplashActivity extends AppCompatActivity {
+
+    private final long VIEW_TIME = 2000;
+
+    /**
+     * Creates the splash screen. Switch to MainActivity after 'VIEW_TIME' milliseconds
+     * @param savedInstanceState The savedInstanceState is a reference to a Bundle object that is passed into the onCreate method of every Android Activity
+     */
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.splash_screen);
+
+        Thread mainThread = new Thread(){
+            @Override
+            public void run() {
+                try {
+                    sleep(VIEW_TIME);
```

Comment:
✏️ This implementation is flawed. If the program loads faster than 2000ms, the user is still forced to watch the splash screen for 2 seconds, regardless of when the program was actually loaded. 

If the program loads slower than 2000ms, then the user watches a splash screen that always goes away at exactly 2000ms, then is forced to stare at a blank screen for the remaining time, no matter how long that could be. 

Solution, don't put the `mainThread` to sleep, let it run simultaneously and have it take over and close the splash screen (kill the splash screen thread) when it is ready to start user interaction.